arch/arm/arm-m/src/arch_nvic.c

```diff
@@ -264,17 +264,23 @@ static void irq_invalid(void)
 int arch_nvic_init(const struct fwk_arch_interrupt_driver **driver)
 {
     uint32_t ictr_intlinesnum;
+#ifndef ARMV6M
```

Comment:
I can see a number of ifdef here, which make the code more difficult to read/follow.
Can you please propose a specific arch_nvic_init_v6m() instead?

Reply:
updated, add additional init function will cause compile error as some instruction used in original init function are not defined for V6M.